### PR TITLE
Trims away the module names on Swift classes

### PR DESCRIPTION
**Pull request by dzenbot**
Port from https://github.com/facebookarchive/ios-snapshot-test-case/pull/219 to new repo.

#### What is this PR about:
Calling `NSStringFromClass` in Objc for Swift classes creates long strings in form of **"Module.Class"**.

To be consistent with the Objc style, specially noticeable on reference paths, this patch trims away the module names on Swift classes.

#### Important:
Merging this could cause tests to fail for users that are currently testing Swift classes, since the image references may not be found.

### CODE REVIEW

**Comment by freak4pc** [comment]
Good change in general, but wouldn't this theoretically break the tests of everyone using it in Swift today? (since the test names would change, so it would require the person to re-record all their snapshots) 

Or am I missing out on something here?

**Comment by alanzeino** [comment]
This fixes an annoyance I also have, but we would definitely have to bundle this in a major version update because we want to follow semantic versioning.

After this change, anyone with tests will have to regenerate (or rename) too, which is unfortunate.

Any chance we can make this opt–in, so that existing tests aren't broken?

**Comment by freak4pc** [comment]
Uhm seems like a relatively simple change (to make opt-in and migrate to single option on next major). Good idea :)

File: FBSnapshotTestCase/FBSnapshotTestCase.m [hunk 1]
```diff
@@ -21,7 +21,8 @@ @implementation FBSnapshotTestCase
 - (void)setUp
 {
   [super setUp];
-  _snapshotController = [[FBSnapshotTestController alloc] initWithTestName:NSStringFromClass([self class])];
+  NSString *testName = [[NSStringFromClass([self class]) componentsSeparatedByString:@"."] lastObject];
```
Comment (nunogoncalves) [inline FBSnapshotTestCase/FBSnapshotTestCase.m]: What about going even further and allow the user to optionally decide the folder's name?

**Comment by nunogoncalves** [comment]
Was just looking into the code to perform this change as well. Any idea when this can get merged? :)

**Comment by alanzeino** [comment]
It can’t be merged until it’s rewritten in a way that doesn’t break existing usage of the library.

**Comment by nunogoncalves** [comment]
What about adding this feature as a opt-in feature, and leave the existing behaviour the default one?  That wouldn't break anything and would not need a major bump, right?

**Comment by alanzeino** [comment]
That’s what I originally requested, yes. But I won’t be making that change on the author’s behalf; waiting on them to do it.

**Comment by nunogoncalves** [comment]
Ah yes, I just saw that, sorry. 

@dzenbot would you be able to do this? If not I might give it a try myself, as this is really annoying. :D

**Comment by nunogoncalves** [comment]
@dzenbot i took the liberty to go ahead and do that in #14. Hope you don’t mind.

**Pull request closed without merging**